README.md

```diff
@@ -147,3 +148,242 @@ If this is not an option (e.g. integration testing), see the following:
 By default, the Compute service account that the VM authorizes as does not have
 k8s engine admin privilege. You need to grant that role to the service account
 via the IAM Admin console.
+
+## Defining app parameters with schema.yaml file
+
+The `schema.yaml` file is the way to declare parameter values that end-user needs to provide
+in order to provision the application. For example, the name of the application, the kubernetes
+namespace, service accounts, etc. It follows a more strict subset of JSON schema specifications.
+The UI uses `schema.yaml` to render the form that end-users will interact with to configure and
+deploy the application.
+
+The information provided by the user for properties defined in `schema.yaml` is available at deployment time. 
+
+This is a simple example of a schema.yaml file:
+
+```yaml
+properties:
+  name:
+    type: string
+    x-google-marketplace:
+      type: NAME
+  namespace:
+    type: string
+    x-google-marketplace:
+      type: NAMESPACE
+required:
+- name
+- namespace
+```
+
+Each entry is defined inside properties, and `required` is a list of required parameters.
+We validate that all required fields are provided before starting the deployment.
+
+### Referencing `schema.yaml` parameter values in kubernetes manifests
+
+All parameters defined in `schema.yaml` can be used in manifests.
+
+#### Helm based deployer
+
+They can be referenced in helm charts just like a regular value from `values.yaml`. 
```

Comment:
We should say that:
- Values from inputs defined by `schema.yaml` will overlay the default `values.yaml` of the chart.
- Dot syntax is supported to overwrite structured values.

For example, the follow `values.yaml`:

```yaml
mysql:
  image: mysql
  replicas: 2
username: admin
...
````

can have a `schema.yaml` covering a subset of values:

```yaml
application_api_version: v1beta1
prroperties:
  name:
    type: string
    x-google-marketplace:
      # Note: This maps to .Release.Name by default
      type: NAME
  namespace:
    type: string
    x-google-marketplace:
      # Note: This maps to .Release.Namespace by default
      type: NAMESPACE
  mysql.image:
    type: string
    default: gcr.io/partner/solution/mysql:5.6
    x-google-marketplace:
      type: IMAGE
  mysql.replicas:
    type: int
    default: 3
```